pkg/smart/check_required_tools_darwin.go

```diff
@@ -7,21 +7,36 @@ import (
 	"bytes"
 	"fmt"
 	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
 )
 
-func checkTools() (string, error) {
-	cmd := exec.Command("/bin/sh", "-c", "smartctl -h")
-	buf := &bytes.Buffer{}
-	cmd.Stdout = bufio.NewWriter(buf)
+const defaultSmartctlPath = "smartctl"
+
+func checkTools(smartctl string) (string, string, error) {
+	cmd := exec.Command("/bin/sh", "-c", fmt.Sprintf("which %s", smartctl))
```

Comment:
This code for getting path of smartctl seems excessive.

What about something like
```go
smartctlPathBuf, err := exec.Command("which", smartctl).Output()
if err != nil {
    return "", "", errors.Wrapf(ErrSmartctlNotFound, "\"%s\"", smartctl)
}
smartctl = strings.TrimRight(string(smartctlPathBuf), "\n")
```

Reply:
Doubt it will work, where needs to run in cmd I think

Reply:
There's really no reason to spawn a shell just to run a command.

I did this test locally while coming up with the comment. It works

```go
package main

import (
    "fmt"
    "os/exec"
)

func main() {
    out, _ := exec.Command("which", "bash").Output()
    fmt.Println(string(out))
}
```

Reply:
Let's consider this in a upcoming PR instead